x/bsonx/bsoncore/array_test.go

```diff
@@ -346,3 +347,120 @@ func TestArray(t *testing.T) {
 		}
 	})
 }
+
+func TestArray_StringN(t *testing.T) {
+	testCases := []struct {
+		description string
+		n           int
+		arr         Array
+		want        string
+	}{
+		// n = 0 cases
+		{"n=0, array with 1 element", 0, Array([]byte{
```

Comment:
Using binary is fairly difficult to verify when reading a test. Suggest making the input a `[]Value` type and then converting that to an array in the test logic:

```go
func TestArray_StringN(t *testing.T) {
	testCases := []struct {
		description string
		n           int
		values      []Value
		want        string
	}{
		{
			description: "n=0, array with 1 element",
			n:           0,
			values: []Value{ // [3.14159]
				{
					Type: TypeDouble,
					Data: AppendDouble(nil, 3.14159),
				},
			},
			want: "",
		},
		{
			description: "n=1, array with 1 element",
			n:           1,
			values: []Value{ // [3.14159]
				{
					Type: TypeDouble,
					Data: AppendDouble(nil, 3.14159),
				},
			},
			want: "[",
		},
	}

	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			got := Array(BuildArray(nil, tc.values...)).StringN(tc.n)
			assert.Equal(t, tc.want, got)
			if tc.n >= 0 {
				assert.LessOrEqual(t, len(got), tc.n)
			}
		})
	}
}
```